src/noob/scheduler.py

```diff
@@ -188,10 +378,7 @@ def _init_graph(nodes: dict[str, NodeSpecification], edges: list[Edge]) -> Topol
             required_edges = [
                 e.source_node
                 for e in edges
-                if e.target_node == node_id
-                and e.required
-                and e.source_node in enabled_nodes
```

Comment:
removed source node check here because a target node should especially not run if its source node is disabled - i.e. we need to keep its dependency on the source node